app/login/register.controller.spec.js

```diff
@@ -0,0 +1,86 @@
+describe('loginController', function () {
+    var element,
+        $rootScope,
+        mockedUsers,
+        userService,
+        mockedNewRegisteredUser,
+        loginController;
+
+    beforeEach(module('app.module', 'foo'));
+
+    beforeEach(inject(function ($compile, $controller, _$rootScope_, $state, $templateCache, _userService_) {
+        $rootScope = _$rootScope_;
+        userService = _userService_;
+
+        mockedUsers = [
+            {
+                email: 'petya',
+                password: 'petya'
+            },
+            {
+                email: 'anna',
+                password: 'anna'
+            },
+            {
+                email: 'goga',
+                password: 'goga'
+            },
+        ];
+
+        mockedNewRegisteredUser = {
+            email: 'Alice',
+            password: 'Alice'
+        };
+
+        loginController = $controller('loginController', {
+            $scope: $rootScope,
+            userService: userService,
+            $state: $state
+        });
+
+        spyOn(userService, 'registerUser').and.returnValue(Promise.resolve(mockedNewRegisteredUser));
+        spyOn(userService, 'loginUser').and.returnValue(Promise.resolve(mockedNewRegisteredUser));
+
+        element = $compile($templateCache.get('login/register-template.html'))($rootScope);
+
+        $('body').append(element);
+
+        $rootScope.$apply();
+    }));
+
+    it('should register user', function () {
+        $('input')[0].value += mockedNewRegisteredUser.email;
```

Comment:
not clear which $('input') your are using here, login or register.

Use helpful classes, like `$(".register-form .email-field")`